classes/Validate.php

```diff
@@ -761,7 +761,9 @@ public static function isZipCodeFormat($zip_code)
      */
     public static function isOrderWay($way)
     {
-        return $way === 'ASC' | $way === 'DESC' | $way === 'asc' | $way === 'desc';
+        $lowerWay = strtolower($way);
+
+        return $lowerWay === 'asc' | $lowerWay === 'desc' | $lowerWay === 'random';
```

Comment:
Hello, I don't think we can do this.

If I am not wrong, `isOrderWay($way)` is used to verify that the provided `$way` can be used in an SQL query using SORT. `random` cannot be used.

In https://github.com/PrestaShop/PrestaShop/blob/develop/src/Adapter/Product/AdminProductDataProvider.php#L195

```
        $sortOrder = Validate::isOrderWay($sortOrder) ? $sortOrder : 'desc';

        $filterParams = $this->combinePersistentCatalogProductFilter(array_merge(
            $post,
            ['last_offset' => $offset, 'last_limit' => $limit, 'last_orderBy' => $orderBy, 'last_sortOrder' => $sortOrder]
        ), $avoidPersistence);
        $filterParams = AdminFilter::sanitizeFilterParameters($filterParams);
```

This block above will behave weird if we modify `Validate::isOrderWay($way)` as you suggest 😉

Reply:
The right word should be `RAND()`

Reply:
@PierreRambaud yes but what i do for that ?

https://github.com/PrestaShop/PrestaShop/blob/develop/src/Core/Product/Search/SortOrder.php#L202-L212

It's complicated and I don't think a BC here is good ?